Adding details document about confirmation pages frontend refactoring (#17742)
MetaMask/metamask-extension: docs/refactoring/confirmation-page-structure/README.md

A lot of other components like `confirm-deploy-contract`, `confirm-send-ether`, `confirm-send-token`, `confirm-approve`, `confirm-contract-interaction` are not doing much and we can get rid of these components and associated routed can also be unified. Check routing document for detail.

## Areas of Refactoring:

1. ### Edit gas popovers:
   There are 2 different versions popovers for gas editing:
   - Legacy gas popover - [component](https://github.com/MetaMask/metamask-extension/tree/develop/ui/components/app/edit-gas-popover)
   - EIP-1559 V2 gas popover - [component1](https://github.com/MetaMask/metamask-extension/tree/develop/ui/components/app/edit-gas-fee-popover), [component2](https://github.com/MetaMask/metamask-extension/tree/develop/ui/components/app/advanced-gas-fee-popover).
     Context [transaction-modal-context](https://github.com/MetaMask/metamask-extension/blob/develop/ui/contexts/transaction-modal.js) is used to show hide EIP-1559 gas popovers.

>      A parent component can be created for gas editing popover which will wrap both the legacy and IEP-1559 gas popover. Depending on the type of transaction appropriate gas popover can be show. `transaction-modal-context` can be used to take care to open/close both the popovers.

Reviewer: ```suggestion
     A parent component can be created for gas editing popover which will wrap both the legacy and EIP-1559 gas popover. Depending on the type of transaction appropriate gas popover can be shown. `transaction-modal-context` can be used to take care to open/close both popovers.
```

   - All warnings related code can be moved to a new child component.
   - Props passing to `confirm-page-component` should be reduced. A lot of passed props like `origin`, `supportEIP1559` can be obtained directly using selectors. Props passing from `confirm-page-container` down to its child components should also be reduced.
4. ### Gas polling:
   Gas polling related code in `/pages/confirm-transaction` can be moved into a hook and included in `pages/confirm-transaction-base`, `/app/token-allowance` as only those confirmation pages need gas estimates.
5. ### Refactor gasFeeInput hook
    [GasFeeInput](https://github.com/MetaMask/metamask-extension/tree/develop/ui/hooks/gasFeeInput) is huge hook exporting about 18 different transaction related values and functions. We need to break this down into smaller hooks and utility functions.

>     The main use o fthis hook is inside [GasFeeContext](https://github.com/MetaMask/metamask-extension/blob/develop/ui/contexts/gasFee.js). Transaction pages are rendered for a transaction, this transaction can is set in this context so that we can avoid passing it over to multiple levels of child components.

Reviewer: ```suggestion
    The main use of this hook is inside [GasFeeContext](https://github.com/MetaMask/metamask-extension/blob/develop/ui/contexts/gasFee.js). Transaction pages are rendered for a transaction, this transaction can is set in this context so that we can avoid passing it over to multiple levels of child components.
```

    - Some of the values like `gasFeeEstimates`, `isNetworkBusy` can be directly obtained from selectors and should be removed from the context.
    - Hooks like `useGasPriceInput`, `useMaxFeePerGasInput`, `useMaxPriorityFeePerGasInput`, etc should be removed from this huge hook and used independent of this.

> Note: This document does not cover signature request pages which are covered separately.

Reviewer: for some reason, this line is being combined with the last bulletpoint. Maybe.. 

```suggestion

**Note:** This document **does not cover signature request pages** which are covered separately.
```

current:
![Screen Shot 2023-02-15 at 8 03 17 PM](https://user-images.githubusercontent.com/20778143/219035473-b56c05e9-cf88-4521-aaed-ff3a6ee5afb2.png)
